earthfile2llb/converter.go

```diff
@@ -477,14 +478,21 @@ func (c *Converter) Volume(ctx context.Context, volumes []string) {
 }
 
 // Env applies the ENV command.
-func (c *Converter) Env(ctx context.Context, envKey string, envValue string) {
+func (c *Converter) Env(ctx context.Context, envKey string, envValue string) error {
 	envValue = c.expandArgs(envValue)
+	invalidEnvKey, err := regexp.MatchString(`[${}()]`, envKey)
```

Comment:
In would go for `[a-zA-Z_]+[a-zA-Z0-9_]*` as you suggested in the PR description. [Source](https://stackoverflow.com/a/2821201).